block/block_postgres.go

```diff
@@ -22,6 +24,47 @@ func (repo *postgresBlockRepository) GetByPos(x, y, z int) (*Block, error) {
 	return entry, err
 }
 
+func (repo *postgresBlockRepository) Iterator(x, y, z int) (chan *Block, error) {
+	rows, err := repo.db.Query(`
+		SELECT posX, posY, posZ, data
+		FROM blocks
+		WHERE posX >= $1 AND posY >= $2 AND posZ >= $3
+		ORDER BY posX, posY, posZ
+		`, x, y, z)
+	if err != nil {
+		return nil, err
+	}
+
+	l := logrus.WithField("iterating_from", []int{x, y, z})
+	ch := make(chan *Block)
+	count := int64(0)
+
+	// Spawn go routine to fetch rows and send to channel
+	go func() {
+		defer close(ch)
+		defer rows.Close()
+
+		l.Debug("Retrieving database rows")
+		for rows.Next() {
+			// Debug progress while fetching rows every 100's
+			count++
+			if count%100 == 0 {
+				l.Debugf("Retrieved %d records so far", count)
+			}
+			// Fetch and send to channel
+			b := &Block{}
+			if err = rows.Scan(&b.PosX, &b.PosY, &b.PosZ, &b.Data); err != nil {
+				l.Errorf("Failed to read next item from iterator: %v", err)
```

Comment:
If error, should break the loop/return; currently it will loop forever if erroring at the end while emitting zero-value blocks.